src/cli.coffee

```diff
@@ -4,55 +4,90 @@
 {
   discover
   reinstall_all
+  shrinkwrap
 }         = require '.'
 cli       = require 'commander'
 path      = require 'path'
+fs        = require 'fs'
 
 cli
   .version '0.0.0' # TODO: Read from package.json
   .description """
     A utility to properly install npm git dependencies.
   """
-  .arguments '<file>'
   .option '-s --silent',  'suppress child processes output'
+  .option '-w, --git_shrinkwrap <path>', 'Optional git shrinkwrap file location'
+  .option '-c, --package <path>', 'Optional git shrinkwrap file location'
   .option '-v --verbose', 'be verbose'
-  .option '-a --all',     'reinstall all git dependencies and devDependencies'
+  #.option '-a --all',     'reinstall all git dependencies and devDependencies'
   .option '-d --dry',     'just print what packages would be installed'
-  .parse process.argv
 
-{
-  silent
-  verbose
-  all
-  dry
-  args: [ file ]
-} = cli
-
-file ?= 'package.json'
-
-if all
-  # TODO: Implement messy, but compatible discovery
-  console.error "I'd love to do it all, but it's not implemented yet. Sorry."
+conf_file_error = (file) ->
```

Comment:
Do we really need this? Both package.json and git_shrinkwrap.json can be required, which is a synchronous operation that will throw if file is not found. It seems like a lot of code just to provide default values and print error message.